src/parser/jobs/drg/modules/Drift.tsx

```diff
@@ -0,0 +1,172 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {Message, Table, Button} from 'semantic-ui-react'
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import {CastEvent} from 'fflogs'
+import Downtime from 'parser/core/modules/Downtime'
+import {getDataBy} from 'data'
+import Module, {dependency} from 'parser/core/Module'
+import React, {Fragment} from 'react'
+import {Timeline} from 'parser/core/modules/Timeline'
+
+// Buffer (ms) to forgive insignificant drift, we really only care about GCD drift here
+// and not log inconsistencies / sks issues / misguided weaving
+const DRIFT_BUFFER = 1200
+
+const DRIFT_ABILITIES = [
+	ACTIONS.HIGH_JUMP.id,
+	ACTIONS.GEIRSKOGUL.id,
+]
+
+const COOLDOWN_MS = {
+	[ACTIONS.HIGH_JUMP.id]: ACTIONS.HIGH_JUMP.cooldown * 1000,
+	[ACTIONS.GEIRSKOGUL.id]: ACTIONS.GEIRSKOGUL.cooldown * 1000,
+}
+
+class DriftWindow {
+	actionId: number
+	start: number
+	end: number = 0
+	drift: number = 0
+	abilityRotation: CastEvent[] = []
```

Comment:
`abilityRotation` and associated methods don't need to be here for DRG's implementation.

Reply:
Er- What do you mean? They're being used later on in the module? If you remove them and the methods, can't really push anything to window?

Reply:
Unless I'm misreading the code, seems like each window only ever has a single ability in it; as soon as you use `addAbility()` and push a HJ or GSK, it closes the window and creates a new one for the next cast. I'm assuming that was yumi's point - that you could replace the array with a single `CastEvent` and just assign it rather than futzing around with adding it and then having to slice in `getLastActionId()`.

Reply:
Oh, I see now. I'll see if I can edit that and fix it up.

Reply:
Or, from that perspective, since all this module cares about is the start & end times of the DriftWindow, you could just assign those based on the appropriate event.timestamp values and not have any ability / abilityRotation (or addAbility method) at all.  Whichever approach makes more sense.

Reply:
Addressed in 301a373